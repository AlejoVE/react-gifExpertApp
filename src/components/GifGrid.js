import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
	const { data: images, loading } = useFetchGifs(category);

	return (
		<>
			<h3 className="animate__animated animate__fadeIn">{category}</h3>
			<div className="card-grid">
				{/* este simbolo && significa que si está cargando, ponga un párrago que diga loading
				Si no está cargando, no hace nada */}
				{loading && <p className="animate__animated animate__flash">Loading...</p>}
				{
					// prettier/ignore
					images.map((img) => (
						<GifGridItem {...img} key={img.id} />
					))
				}
			</div>
		</>
	);
};

GifGrid.propTypes = {
	category : PropTypes.string.isRequired,
}