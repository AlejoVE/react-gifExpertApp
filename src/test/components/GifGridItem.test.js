import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Tests in <GifGridItem/> component', () => {
	const url = 'http://localhost:8080/mypagina.com';
	const title = 'My tittle';
	let wrapper = shallow(<GifGridItem url={url} title={title} />);

	test('should show <GifGridItem /> component correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should show a title "My title"', () => {
		const pEl = wrapper.find('p').text();
		expect(pEl).toBe(title);
	});

	test('should have an image with the given url ', () => {
		const imgEl = wrapper.find('img');
		const srcProp = imgEl.prop('src');

		expect(srcProp).toBe(url);
		expect(imgEl.prop('alt')).toBe(title);
	});

	test('should have a div with the a className ', () => {
		const divEl = wrapper.find('div');
		const className = divEl.prop('className');

		expect(className.includes('animate__fadeIn')).toBe(true);
		// expect(className).toBe(
		// 	'card animate__animated animate__fadeIn animate__fast 800ms'
		// );
	});
});
