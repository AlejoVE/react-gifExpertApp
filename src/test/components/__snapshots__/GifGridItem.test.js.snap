// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItem/> component should show <GifGridItem /> component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__fast 800ms"
>
  <img
    alt="My tittle"
    src="http://localhost:8080/mypagina.com"
  />
  <p>
    My tittle
  </p>
</div>
`;
